Sources/EpoxyCollectionView/CollectionView/CollectionView.swift

```diff
@@ -811,29 +851,22 @@ extension CollectionView: UICollectionViewDelegate {
     didEndDisplaying cell: UICollectionViewCell,
     forItemAt indexPath: IndexPath)
   {
-    // When updating, items ending display correspond to items in the old data.
-    let data: CollectionViewData?
-    switch updateState {
-    case .notUpdating, .preparingUpdate:
-      data = epoxyDataSource.data
-    case .updating(from: let oldData):
-      data = oldData
+    guard let cell = cell as? CollectionViewCell else {
+      EpoxyLogger.shared.assertionFailure("Cell does not match expected type CollectionViewCell.")
+      return
     }
 
+    let data = data(for: cell, at: indexPath)
+
     guard
-      let item = data?.itemIfPresent(at: indexPath),
-      let section = data?.sectionIfPresent(at: indexPath.section)
+      let item = data?.item(at: indexPath),
```

Comment:
Notice how the `willDisplay` callback didn't call the `ifPresent` variant of this method. I believe this is because someone ran into this issue I just fixed in the past, but decided to fix it by calling the non-asserting function (the ifNeeded variant).

With this more robust fix, we can bring this in parity with `willDisplay` and add a helpful assert back.